relay-server/src/services/buffer/mod.rs

```diff
@@ -443,6 +410,128 @@ impl EnvelopeBufferService {
         }
     }
 
+    async fn pop_and_forward<'a>(
+        partition_tag: &str,
+        services: &Services,
+        buffer: &mut PolymorphicEnvelopeBuffer,
+        project_key_pair: ProjectKeyPair,
+    ) -> Result<(), EnvelopeBufferError> {
+        let mut at_least_one_pending = false;
+
+        let own_key = project_key_pair.own_key();
+        let own_project = services.project_cache_handle.get(own_key);
+        if let ProjectState::Pending = own_project.state() {
```

Comment:
```suggestion
        if own_project.state().is_pending() {
```

reads better imo, feel free to ignore